platform/mbed_enum_flags.h

```diff
@@ -0,0 +1,167 @@
+/*
+ * Copyright (c) 2020 ARM Limited. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ * Licensed under the Apache License, Version 2.0 (the License); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an AS IS BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef MBED_SCOPED_ENUM_FLAGS_H
+#define MBED_SCOPED_ENUM_FLAGS_H
+
+#include <type_traits>
+
+#define ENUM_FLAG_OPERATOR(T, OP) \
+inline constexpr T operator OP( T lhs, T rhs) \
+{ \
+    return T (static_cast<std::underlying_type_t< T > >(lhs) OP \
+        static_cast<std::underlying_type_t< T > >(rhs)); \
+}
+
+
+#define ENUM_FLAG_OPERATOR_LHS_REFERNCE(T, OP) \
+inline constexpr T & operator OP (T & a, T b) \
+{ \
+    return a = a OP b; \
+}
+
+
+/**
+ * @brief Applies bitwise operators to a enum class defined elsewhere.
+ *
+ * @param T             The enum class typename
+ *
+ * This macro applies the bitwise negate, AND, OR, XOR operators and the
+ * corresponding assignment operators to an existing enum class. The macro
+ * uses underlying type traits to convert back and forth.
+ *
+ * Usage:
+ * @code
+ * external_enum.h:
+ *
+ * enum class SpokenLanguages : uint8_t {
+ *     Sindarin    = (1 << 0),
+ *     Westron     = (1 << 1),
+ *     Rohirric    = (1 << 2),
+ *     BlackSpeech = (1 << 3)
+ * };
+ *
+ * my_code.cpp:
+ *
+ * #include "mbed_enum_flags.h"
+ * #include "external_enum.h"
+ *
+ * MBED_ENUM_FLAG_APPLY_OPERATORS(SpokenLanguages)
+ *
+ * SpokenLanguages gandalf = SpokenLanguages::Sindarin | SpokenLanguages::Westron |
+ *     SpokenLanguages::Rohirric | SpokenLanguages::BlackSpeech;
+ * @endcode
+ *
+ */
+#define MBED_ENUM_FLAG_APPLY_OPERATORS(T) \
+    inline constexpr T operator ~( T lhs) \
+    { \
+        return ( T ) (~static_cast<std::underlying_type_t< T > >(lhs)) ; \
```

Comment:
Doesn't match the constructor form of the binary op now. More excess spaces. (Don't need spaces to separate `>>` any more.

Reply:
> Doesn't match the constructor form of the binary op now.

Oops!

> More excess spaces. (Don't need spaces to separate `>>` any more.

I was being paranoid. I figured working was better than pretty. I was not sure I would not cause it to implode if I removed spaces. I was sure if I did have spaces and it worked with gcc, it should work for each supported compiler.

Reply:
`>>` is special - there needs to be a magic "tokensiation" rule that gets it treated as `> >` in templates. That was added as a bit of a hack in C++11, but it's standard.